htdocs/core/class/html.form.class.php

```diff
@@ -2114,7 +2119,7 @@ public function select_produits_list($selected = '', $htmlname = 'productid', $f
 		}
         // Units
         if (! empty($conf->global->PRODUCT_USE_UNITS)) {
-            $sql .= ', u.label as unit_long, u.short_label as unit_short';
+            $sql .= ", u.label as unit_long, u.short_label as unit_short, p.weight, p.weight_units, p.length, p.length_units, p.width, p.width_units, p.height, p.height_units, p.surface, p.surface_units, p.volume, p.volume_units";
             $selectFields .= ', unit_long, unit_short';
```

Comment:
Also add new SELECT field to $selectFields else you get groupby sql error

Reply:
I fixed selectFields for group by, thanks you.